### PR TITLE
delete methods to remove ambiguities

### LINKED ISSUE #16: method ambiguity in convert
```julia
using SumOfSquares, DynamicPolynomials, SemialgebraicSets
@polyvar x
model = SOSModel()
@constraint(model, x^2 >= 0, domain = @set(x >= 0))
```
triggers
```
ERROR: MethodError: convert(::Type{BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}}, ::BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}) is ambiguous. Candidates:
  convert(::Type{BasicSemialgebraicSet{T,PT,AT}}, set::BasicSemialgebraicSet{T,PT,AT}) where {T, PT<:AbstractPolynomialLike, AT<:AbstractAlgebraicSet} in SemialgebraicSets at /home/thoth/.julia/packages/SemialgebraicSets/wN4gl/src/basic.jl:24
  convert(::Type{BasicSemialgebraicSet{T,PT,AT}}, set::BasicSemialgebraicSet) where {T, PT, AT} in SemialgebraicSets at /home/thoth/.julia/packages/SemialgebraicSets/wN4gl/src/basic.jl:27
Stacktrace:
 [1] SumOfSquares.SOSPolynomialSet{BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace},Monomial{true},MonomialVector{true},SumOfSquares.Certificate.Putinar{SumOfSquares.Certificate.Newton{SOSCone,MonomialBasis,Tuple{}},SOSCone,MonomialBasis}}(domain::BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}, monomials::MonomialVector{true}, certificate::SumOfSquares.Certificate.Putinar{SumOfSquares.Certificate.Newton{SOSCone,MonomialBasis,Tuple{}},SOSCone,MonomialBasis})
   @ SumOfSquares ~/.julia/packages/SumOfSquares/rsQiW/src/sos_polynomial.jl:33
 [2] moi_set(cone::SOSCone, monos::MonomialVector{true}; domain::BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}, basis::Type, newton_polytope::Tuple{}, maxdegree::Int64, sparsity::NoSparsity, newton_of_remainder::Bool, ideal_certificate::SumOfSquares.Certificate.Newton{SOSCone,MonomialBasis,Tuple{}}, certificate::SumOfSquares.Certificate.Putinar{SumOfSquares.Certificate.Newton{SOSCone,MonomialBasis,Tuple{}},SOSCone,MonomialBasis})
   @ SumOfSquares ~/.julia/packages/SumOfSquares/rsQiW/src/constraint.jl:131
 [3] build_constraint(_error::Function, p::Polynomial{true,Float64}, cone::SOSCone; kws::Base.Iterators.Pairs{Symbol,BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace},Tuple{Symbol},NamedTuple{(:domain,),Tuple{BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}}}})
   @ SumOfSquares ~/.julia/packages/SumOfSquares/rsQiW/src/constraint.jl:167
 [4] add_constraint(model::Model, constraint::PolyJuMP.Constraint{JuMP.var"#_error#68"{Symbol},Polynomial{true,Float64},PolyJuMP.NonNegPoly,Base.Iterators.Pairs{Symbol,BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace},Tuple{Symbol},NamedTuple{(:domain,),Tuple{BasicSemialgebraicSet{Int64,Polynomial{true,Int64},FullSpace}}}}}, name::String)
   @ PolyJuMP ~/.julia/packages/PolyJuMP/KipJi/src/constraint.jl:214
 [5] top-level scope
   @ ~/.julia/packages/JuMP/YXK4e/src/macros.jl:417
```
on SumOfSquares master, SemialgebraicSets v0.2.1

edit: on Julia 1.6

### PR DESCRIPTION
**Pull request by lkapelevich**
fixes #16 
I added the method that was "suggested by Julia" when the ambiguity error is thrown, though I don't actually understand why we are going into `convert` from places like https://github.com/jump-dev/SumOfSquares.jl/blob/master/src/sos_polynomial.jl#L46, or if this should be considered a Julia issue?
This should probably be tested on more Julia versions in the CI scripts.

### CODE REVIEW

**Comment by chriscoey** [comment]
Yes this issue has been bugging me for a while

File: src/basic.jl [hunk 1]
```diff
@@ -16,14 +16,9 @@ function BasicSemialgebraicSet(V::AlgebraicSet{T, PT, A, ST}, p::Vector{PS}) whe
     BasicSemialgebraicSet(convert(AlgebraicSet{U, PU, A, ST}, V), Vector{PU}(p))
 end
 #BasicSemialgebraicSet{T, PT<:APL{T}}(V::AlgebraicSet{T, PT}, p::Vector{PT}) = BasicSemialgebraicSet{T, PT}(V, p)
-function basicsemialgebraicset(V, p)
```
Comment (blegat) [inline src/basic.jl:19]: Why removing this ? There is no need to introduce this breaking change

Comment (lkapelevich) [reply src/basic.jl:19]: Sorry, I didn't notice it was exported. It didn't get hit in tests.

**Comment by lkapelevich** [comment]
@blegat was there anything else?

**Review by blegat — APPROVED** [review]

**Pull request merged**